src/domains/idpay-app/api/idpay_wallet/openapi.wallet.yml.tpl

```diff
@@ -0,0 +1,689 @@
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+        '429':
+          description: Too many Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+        '500':
+          description: Server ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+  /instrument:
+    put:
+      tags:
+        - wallet
+      summary: Association of a payment instrument to an initiative
+      operationId: enrollInstrument
+      requestBody:
+        description: 'Unique identifier of the subscribed initiative, instrument HPAN'
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/InstrumentPutDTO'
+            example:
+              initiativeId: string
+              hpan: string
+      responses:
+        '200':
+          description: Enrollment OK
+          content:
+            application/json: { }
+        '400':
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+        '401':
+          description: Authentication failed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+        '403':
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+        '404':
+          description: The requested ID was not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+        '429':
+          description: Too many Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+        '500':
+          description: Server ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+              example:
+                code: 0
+                message: string
+  
+  /{initiativeId}/instruments/{hpan}:              
```

Comment:
@pp-ps 
HPAN come path parameter?

Reply:
HPAN must be superseded by a new random ID, coming from the PM, especially if this is an user-initiated flow.
HPAN could stay in a path param, however it is set to be replaced anyway.

cc @GiovanniMancini

Reply:
@francescopersico @pp-ps  Thanks for your comments, I understand your point and share your perplexity.
In order to address it, I'm evaluating the use of `idWallet` as payment instrument id, which _is_ a random id generated by Payment Manager. As far as I know, the same payment instrument can be associated with different `idWallet`s in its lifetime, thus the `idWallet` shouldn't be used to identify a payment instrument resource in a Rest API, since it's not unique. I'm trying to understand when the `idWallet` changes and if/when IO App receives it from PM